include/biosHandler.hpp

```diff
@@ -4,16 +4,81 @@
 
 namespace vpd
 {
+
+/**
+ * @brief Interface class for BIOS handling.
+ *
+ * The class layouts virtual methods required to be implemented by any concrete
+ * class intended to use feature provided via BIOS handler class.
+ */
+class BiosHandlerInterface
+{
+  public:
+    /**
+     * @brief API to back up or restore BIOS attributes.
+     *
+     * Concrete class should implement the API and read the backed up data from
+     * its designated location and take a call if it should be backed up or
+     * restored.
+     */
+    virtual void backUpOrRestoreBiosAttributes() = 0;
+
+    /**
+     * @brief API to be triggered on BIOS attribute change.
+     *
+     * Concrete class should implement the API to capture any change in the BIOS
```

Comment:
is it capturing the changes or this API will be triggered when any changes caught ?
or it handles listening and handling the code on changes ?

Reply:
Ok, yes I agree, this will not listen it will be triggered based on event. I will update the comment.